framework/scripts/agent_upgrade.py

```diff
@@ -90,17 +89,36 @@ def get_agents_versions(agents: list) -> dict:
     dict
         Dictionary with the current version (prev_version).
     """
-    agents_versions = dict()
-    for agent_id in agents:
-        agent = Agent(agent_id)
-        agent.load_info_from_db()
-        if agent.version:
-            agents_versions[agent_id] = {
-                'prev_version': agent.version,
-                'new_version': None
-            }
+    f_kwargs = {
+        "agent_list": agents,
+        "select": ["version"],
+        "limit": len(agents)
+    }
+    agent_versions = await cluster_utils.forward_function(get_agents, f_kwargs=f_kwargs)
+    return {agent['id']: {"prev_version": agent['version'], "new_version": None}
+            for agent in agent_versions.affected_items}
 
-    return agents_versions
+
+async def get_agent_version(agent_id: str) -> str:
```

Comment:
Can we have just 1 function to get the agent/s version? It could accept a list of any length, including 1 element only.

Reply:
Both functions return different data structures. It is achievable but it will require further logic to parse the result. It's up to you.